### ISSUE TITLE
JAX/Flax NNX implementation of SimpleFold

### ISSUE THREAD

**Issue by SjoerdWiarda** [apple/ml-simplefold#28]
Hi all,

I recently implemented SimpleFold using `JAX`/`Flax NNX`. The implementation supports loading in all `PyTorch` weights for all sizes/variants (including the confidence/pLDDT module) and running inference with the `JAX` model. 
The same `inference.py` script (or `sample.ipynb`) as used for the original `SimpleFold` can be used. The option `--backend` has been extended to include `jax`. 

The repository can be found here: https://github.com/SjoerdWiarda/ml-simplefold-jax.

To make sure that the accuracy/output is equivalent between the `PyTorch` and `JAX` model, I added several regression tests. When evaluating the modules separately (for example the Folding Diffusion Transformer, or the ESM model) they match almost perfectly (up to numerical precision). 

I plan to add some more regression checks, but I believe I removed all deltas. For example, I changed the `LayerNorm` epsilon from the `NNX` default of `1e-6` to the `PyTorch` default of `1e-5`, or I changed the use of the exact GELU for the ESM model (the folding model uses the approximate version). 

Most of my development was done on my local computer with insufficient VRAM. For this reason I have not been able to benchmark speed/runtime differences between backends.

I hope this implementation is useful to others, feel free to use it. If there are any questions/suggestions, feel free to ask them/contact me.
